### PR TITLE
Adding tests for proc util

### PR DESCRIPTION
**Pull request by IanWoodard**
Beginning to add testing for our proc utility

### CODE REVIEW

File: tests/proc/test_quote.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from __future__ import annotations
+
+from devenv.lib.proc import quote
+
+
+# Sanity check
+def test_quote() -> None:
```
Comment (joshuarli) [inline tests/proc/test_quote.py:7]: don't think this is really useful / testing much of anything

File: tests/proc/test_xtrace.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from __future__ import annotations
+
+from pytest import CaptureFixture
+
+from devenv.lib.proc import xtrace
+
+
+# Sanity check
+def test_xtrace(capfd: CaptureFixture[str]) -> None:
```
[the remark this answers is not in the document]
Comment (joshuarli) [reply tests/proc/test_xtrace.py:9]: There's an overhead per test to be paid and it's preferable to have a better single test than multiple trivial ones.

File: tests/proc/test_run.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        mock_run.side_effect = FileNotFoundError("Command not found")
+        with pytest.raises(SystemExit):
+            run(cmd, exit=True)
+
+    mock_run.assert_called_once_with(
+        cmd, check=True, stdout=None, cwd=None, env=env
+    )
+
+
+def test_run_with_custom_env() -> None:
```
[the remark this answers is not in the document]
Comment (joshuarli) [reply tests/proc/test_run.py:82]: Like if we have ("echo", "$PATH") we should assert for the expected stdout.

**Comment by joshuarli** [comment]
```
==> Downloading https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/mac-x64/chromedriver-mac-x64.zip
==> Installing Cask chromedriver
Warning: macOS's Gatekeeper has been disabled for this Cask
==> Purging files for version 119.0.6045.105 of Cask chromedriver
Error: It seems the symlink source '/usr/local/Caskroom/chromedriver/119.0.6045.105/chromedriver-mac-x64/chromedriver' is not there.
Installing chromedriver has failed!
Homebrew Bundle failed! 1 Brewfile dependency failed to install.
Traceback (most recent call last):
```

ugh i think we should just mock some of this brew stuff

**Comment by IanWoodard** [comment]
> ugh i think we should just mock some of this brew stuff

This is unrelated to the change, right?

File: tests/proc/test_run.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        cmd, check=True, stdout=None, cwd=None, env=env
+    )
+
+
+def test_run_command_not_found() -> None:
+    cmd = ("invalid_command",)
+    user_environ = {"PATH": "/usr/local/bin:/usr/bin"}
+    env = {**user_environ, **base_env}
+
+    with patch("devenv.lib.proc.subprocess_run") as mock_run, patch(
```
Comment (joshuarli) [inline tests/proc/test_run.py:70]: can we actually run subprocess_run wherever possible? i didn't mean to only change that one test, i meant it generally

**Pull request merged**